core/schema/utils.go

```diff
@@ -0,0 +1,21 @@
+package schema
+
+import "fmt"
+
+func getNonEmpty(args ...string) string {
```

Comment:
Might not be related with this PR but having a function name `getNonEmpty` and at some point it is possible to get empty value seems counter intuitive, I think better to rename it or just return error if it is possible.

Reply:
+1, we can cover it another PR.